src/sidebar/services/auth.js

```diff
@@ -0,0 +1,311 @@
+import { TinyEmitter } from 'tiny-emitter';
+
+import serviceConfig from '../config/service-config';
+import OAuthClient from '../util/oauth-client';
+import { resolve } from '../util/url';
+
+/**
+ * @typedef {import('../util/oauth-client').TokenInfo} TokenInfo
+ *
+ * @typedef RefreshOptions
+ * @property {boolean} persist - True if access tokens should be persisted for
+ *   use in future sessions.
+ */
+
+/**
+ * Authorization service.
+ *
+ * This service is responsible for acquiring access tokens for making API
+ * requests and making them available via the `tokenGetter()` method.
+ *
+ * Access tokens are acquired via the OAuth authorization flow, loading valid
+ * tokens from a previous session or, on some websites, by exchanging a grant
+ * token provided by the host page.
+ *
+ * Interaction with OAuth endpoints in the annotation service is delegated to
+ * the `OAuthClient` class.
+ *
+ * @inject
+ */
+export class AuthService extends TinyEmitter {
+  /**
+   * @param {Window} $window
+   * @param {import('./api-routes').APIRoutesService} apiRoutes
+   * @param {import('./local-storage').LocalStorageService} localStorage
+   * @param {import('./toast-messenger').ToastMessengerService} toastMessenger
+   */
+  constructor($window, apiRoutes, localStorage, settings, toastMessenger) {
+    super();
+
+    /**
```

Comment:
The code from here until the end of the constructor is taken from the body of the old `auth` function without changes.